### ISSUE TITLE
Weird behaviour with meteor accounts login

### ISSUE THREAD

**Issue by serkandurusoy** [TelescopeJS/Telescope#770]
Today I wanted to post a comment on a crater.io post from a new computer. Obviously, I was not logged in.

I got presented wih options to sign in and since I did not remember what I used to sign up to crater.io (I have almost always been already logged in to crater.io for as long as I can remember) so among the presented choices I though, I must have signed up with meteor accountd (doh!)

As soon as I logged in to my meteor account I realised I had not used it for crater io login which asked me for an email address for completing the signup. Well, I could not enter my original email address with which I had already signed up, so I had to fake a new user name and email address because I did not have any other option to discard this new user, sign out, sign in with another user etc. I just got stuck until I entered a non-existing user credentials for a new user!

So to summarize, telescope should be able to match my email address to an existing account and merge as necessary. 

Now I have two users on crater.io, serkandurusoy and serkandurusoy0

**Comment by splendido** [comment]
See also #758

**Comment by serkandurusoy** [comment]
In order to improve user experience in the meantime, is it possible to set the workflow up in such a way that a new user can discard the signup processs?

In my case, it was frustrating to have gotten stuck at a prompt that asks for a username and an email address with no way to cancel or log out.

Furthermore, what's the point of using an oauth provider if I'm going to be asked explicitly for a username and email address anyway!

**Comment by SachaG** [comment]
There isn't really any good way to address this right now I'm afraid. The process you experienced isn't great, but it's kind of what we're stuck with for now…

**Comment by serkandurusoy** [comment]
I do understand the situation, but as much as #758 is a feature, this one is definitely a bug and should be kept open either to be resolved or referenced.

This use case literally blocks the whole UI and forces the user to either clear localstorage or fake a username/email combo.

Inexperienced users may not be able to think of doing either, rendering the whole telescope based site useless to a group of *existing users* of that site who have already invested time and effort in signing up with that site.

So I think this is a pretty important, blocking bug that should be reopened and given more thought.

If it cannot be resolved, I believed it at least deserves a big red warning label.

**Comment by SachaG** [comment]
OK, I'll look into it. I thought you could at least log out.

**Comment by splendido** [comment]
The thing is that after the login process throug OAuth is successfully completed, Telescope promts the user to add `username` and `email` (if I correctly understood...).

At that point the new user was already created and logged in: *cancelling* the registration would mean also removing the newly created user from the DB.

**Comment by serkandurusoy** [comment]
@splendido yes that is correct and possibly the intended course of action. if a user clicks on signin/up with oauth provider and gets propted to enter new user credentials only to find out that their credentials do exist on that site, they have two options:

a) merge accounts (#758)

or

b) cancel, either try some passwords for that username or retrieve the forgotten password

since we are not yet able to merge the accounts, the best course of action would be to 

delete the already created user and log them out (which deletion would probably cause automatically). And maybe also present them with a friendly message explaining the situation.

**Comment by xavcz** [comment]
Hey folks! 

Even if this issue is still topical, we are slowly *demeteorizing* Telescope (React / Apollo / Webpack). I think we won't investigate more on this merging or delete already created user, even if these solutions are "viable". We'll be able to fix this kind of issue in the future with the account solution we'll use 👍 (passport maybe?) 

If you are interested in joining the effort to make it works with Apollo / Webpack, come by the Slack room and checkout on `apollo` branch (messy at the moment, but in progress! 🚀 )

I'm closing it.
